src/navigation/pathcurve.cpp

```diff
@@ -57,21 +69,20 @@ std::vector<glm::dvec3> PathCurve::points() const {
 
 void PathCurve::initializeParameterData() {
     _nSegments = static_cast<int>(_points.size() - 3);
-
     ghoul_assert(_nSegments > 0, "Cannot have a curve with zero segments!");
 
     _curveParameterSteps.clear();
     _lengthSums.clear();
     _parameterSamples.clear();
 
+    const double max = static_cast<double>(_nSegments);
+
     // Evenly space out parameter intervals
     _curveParameterSteps.reserve(_nSegments + 1);
-    const double dt = 1.0 / _nSegments;
-    _curveParameterSteps.push_back(0.0);
-    for (unsigned int i = 1; i < _nSegments; i++) {
-        _curveParameterSteps.push_back(dt * i);
+    for (double i = 0.0; i < max; i += 1.0) {
```

Comment:
Why the iteration in double?
```
for (int i = 0; i < _nSegments; i++) {
    _curveParameterSteps.push_back(static_cast<double>(i));
}
```
would read less weird and no need for the `max` variable